src/base/static/components/organisms/main-map.js

```diff
@@ -397,22 +406,39 @@ class MainMap extends Component {
 
     for (let layerId in this.props.layerStatuses) {
       if (
-        this.props.layerStatuses[layerId].isVisible &&
-        !prevProps.layerStatuses[layerId].isVisible
-      ) {
-        // A layer has been switched on.
-        this.addLayer(
-          this.props.layers.find(layer => layer.id === layerId),
-          this.props.layerStatuses[layerId].isBasemap,
-        );
-      } else if (
-        !this.props.layerStatuses[layerId].isVisible &&
+        this.props.layerStatuses[layerId].isVisible !==
         prevProps.layerStatuses[layerId].isVisible
       ) {
-        // A layer has been switched off.
-        this._map.removeLayer(
-          this.props.layers.find(layer => layer.id === layerId),
-        );
+        if (this.props.layerStatuses[layerId].isVisible) {
+          // A layer has been switched on.
+          this.addLayer(
+            this.props.layers.find(layer => layer.id === layerId),
+            this.props.layerStatuses[layerId].isBasemap,
+          );
+        } else {
+          // A layer has been switched off.
+          this._map.removeLayer(
+            this.props.layers.find(layer => layer.id === layerId),
+          );
+        }
+      }
+
+      if (
```

Comment:
On this branch if you turn a layer on, then off, then on again the layer will hang. If you repeat the process the layer will show correctly. It seems like every other time you switch on a layer it will render on the map.

I feel like I'm still missing something about the goal of this PR, but I found that if you remove lines 426-441 everything works correctly. 

Why are we calling `addLayer` if the status changes to `loaded`? The `loaded` state is triggered by mapboxgl once the layer has loaded after a call to `addLayer`, so I think there's a circular thing going on here.

If we just remove lines 426-441 is the decoupling goal of this PR still met?

Reply:
>Why are we calling addLayer if the status changes to loaded? The loaded state is triggered by mapboxgl once the layer has loaded after a call to addLayer, so I think there's a circular thing going on here.

Ah, I see - I think I was missing the concept of what "loaded" means. I'm going to think on this for a bit...

The goal of this PR is to decouple the "status" from the "isVisible" state. I think the mapping code is more complicated than it needs to be, and I think it's worthwhile to straighten it out a bit instead of building on the complexity.

For example, the `isVisible: layer.type !== "place" ? !!layer.is_visible_default : false,` in the Map duck on the "simplify layer handling" PR is what raised the red flag for me.